src/main/java/gateways/JavaHttpGateway.java

```diff
@@ -89,7 +95,7 @@ public String createFilterOptions(String mealType, String calories, String time)
         }
 
         if (convertCookTime.containsKey(time)) {
-            filterFields.append(convertCookTime.get(mealType));
+            filterFields.append(convertCookTime.get(time));
```

Comment:
is this what was causing the issue in the filtered search?

Reply:
This was the issue.